OSDOCS-12732: Documented the 4.17.6 z-stream release notes (#85277)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

=== RHBA-2024:10137 - {product-title} {product-version}.6 bug fix, and security update advisory

Issued: 26 November 2024

{product-title} release {product-version}.6 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2024:10137[RHBA-2024:10137] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2024:10140[RHBA-2024:10140] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.17.6 --pullspecs
----

[id="ocp-4-17-6-enhancements_{context}"]
==== Enhancements

[id="ocp-4-17-6-kubernetes-1-30-6_{context}"]
===== Updating to Kubernetes version 1.30.6

{product-title} release {product-version}.6 contains the changes that come from the update to Kubernetes version 1.30.6. (link:https://issues.redhat.com/browse/OCPBUGS-44512[*OCPBUGS-44512*])

[id="ocp-4-17-6-bug-fixes_{context}"]
==== Bug fixes

//OCPBUGS-44760 Need to confirm with SME.

> * Previously, if you set custom annotations in a custom resource (CR), the SR-IOV Operator would override all the default annotations in the `SriovNetwork` CR. With this release, when you define custom annotations in a CR, the SR-IOV Operator  does not override the default annotations. (link:https://issues.redhat.com/browse/OCPBUGS-42252[*OCPBUGS-42252*])

Reviewer: ```suggestion
* Previously, if you set custom annotations in a custom resource (CR), the SR-IOV Operator would override all the default annotations in the `SriovNetwork` CR. With this release, when you define custom annotations in a CR, the SR-IOV Operator does not override the default annotations. (link:https://issues.redhat.com/browse/OCPBUGS-42252[*OCPBUGS-42252*])
```

* Previously, in the Red{nbsp}Hat {product-title} web console *Notifications* section, silenced alerts were visible in the notification drawer because the alerts did not include external labels. With this release, the alerts include external labels so that silenced alerts are not visible on the notification drawer. (link:https://issues.redhat.com/browse/OCPBUGS-44722[*OCPBUGS-44722*])

* Previously, when you clicked the `start lastrun` option in the Red{nbsp}Hat {product-title} web console *Edit BuildConfig* page, an error prevented the `lastrun` operation from running. With this release, a fix ensures that `start lastrun` option runs as expected. (link:https://issues.redhat.com/browse/OCPBUGS-44587[*OCPBUGS-44587*])

* Previously, {product-title} {product-version} introduced collapsing and expanding the *Getting started* section on the *Administrator perspective* of the Red{nbsp}Hat {product-title} web console. When you either collapsed or expanded the section, you could not close the section. With this release, a fix ensures that you can now close the *Getting started* section. (link:https://issues.redhat.com/browse/OCPBUGS-44586[*OCPBUGS-44586*])

> * Previously, the `MachineConfig` tab on the *Details* page of the Red{nbsp}Hat {product-title} web console displayed an error when tab displayed an error when one or more `spec.config.storage.files` did not include data fields that were marked as optional. With this update, a fix ensures that this error no longer displays if you populated no values in the optional fields. (link:https://issues.redhat.com/browse/OCPBUGS-44479[*OCPBUGS-44479*])

Reviewer: This don't sound right! 

>  web console displayed an error when tab displayed an error when one or more `spec.config.storage.files` did not include data fields that were marked as optional.

Reviewer: Maybe? 
```suggestion
* Previously, the `MachineConfig` tab on the *Details* page of the Red{nbsp}Hat {product-title} web console displayed an error when one or more `spec.config.storage.files` did not include data fields that were marked as optional. With this update, a fix ensures that this error no longer displays if you populated no values in the optional fields. (link:https://issues.redhat.com/browse/OCPBUGS-44479[*OCPBUGS-44479*])
```

> * Previously, a {hcp-capital} cluster that used the {ibm-name} platform was unable to accept authenticatation from an `oc login` command. This behavior caused a an error for the web broswer where the browser could not fetch the token from the cluster. With this release, a fix ensures that cloud-based endpoints do not get proxied so that authentication by using the `oc login` command works as expected. (link:https://issues.redhat.com/browse/OCPBUGS-44276[*OCPBUGS-44276*])

Reviewer: ```suggestion
* Previously, a {hcp-capital} cluster that used the {ibm-name} platform was unable to accept authenticatation from an `oc login` command. This behavior caused an error for the web broswer where the browser could not fetch the token from the cluster. With this release, a fix ensures that cloud-based endpoints do not get proxied so that authentication by using the `oc login` command works as expected. (link:https://issues.redhat.com/browse/OCPBUGS-44276[*OCPBUGS-44276*])
```

// SME: 

> * Previously, if `RendezvousIP` matched a substring in the `next-hop-address` field of a compute node configuration, a validation error occured and outputted an error message. The `RendezvousIP` must only match a control plane host address. With this release, a substring comparison for `RendezvousIP` is used only against a control plane host address, so that the error no longer exists. (link:https://issues.redhat.com/browse/OCPBUGS-44261[*OCPBUGS-44261*])

Reviewer: Not sure we need this phrase. 
```suggestion
* Previously, if the `RendezvousIP` matched a substring in the `next-hop-address` field of a compute node configuration, a validation error occurred. The `RendezvousIP` must match only a control plane host address. With this release, a substring comparison for `RendezvousIP` is used only against a control plane host address, so that the error no longer exists. (link:https://issues.redhat.com/browse/OCPBUGS-44261[*OCPBUGS-44261*])
```
